File: project/hyeok-onboarding/src/main/java/com/INNER_CIRCLE_ICD4/innerCircle/domain/Choice.java
package com.INNER_CIRCLE_ICD4.innerCircle.domain;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Choice {

    @Id
    @GeneratedValue
    private UUID id;

    private String text;
    private int choiceIndex;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "question_id")
    @JsonBackReference // ✅ 순환 참조 방지
    private Question question;

    public Choice(String text, int choiceIndex) {
        this.text = text;
        this.choiceIndex = choiceIndex;
        this.createdAt = LocalDateTime.now();
        this.updatedAt = LocalDateTime.now();
    }

    public Choice(String text, Question question) {
        this.text = text;
        this.question = question;
        this.createdAt = LocalDateTime.now();
        this.updatedAt = LocalDateTime.now();
    }

    public void updateText(String newText) {
        this.text = newText;
        this.updatedAt = LocalDateTime.now();
    }

    public void setQuestion(Question question) {
        this.question = question;
    }
}

File: project/hajin-onboarding/src/main/kotlin/com/innercircle/survey/survey/domain/Choice.kt
package com.innercircle.survey.survey.domain

import com.innercircle.survey.common.domain.BaseEntity
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.Table

@Entity
@Table(name = "CHOICES")
class Choice private constructor(
    @Column(name = "text", nullable = false, length = 200)
    var text: String,
) : BaseEntity() {
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "question_id")
    var question: Question? = null
        internal set

    fun updateText(text: String) {
        require(text.isNotBlank()) { "선택지 텍스트는 필수입니다." }
        this.text = text
    }

    companion object {
        fun create(text: String): Choice {
            require(text.isNotBlank()) { "선택지 텍스트는 필수입니다." }
            return Choice(text)
        }
    }
}
